multiversx_sdk_cli/args_validation.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any
+
+from multiversx_sdk_cli.errors import InvalidArgumentsError
+
+
+def ensure_required_transaction_args_are_provided(args: Any):
+    _ensure_nonce_args(args)
+    _ensure_receiver_args(args)
+    _ensure_gas_limit_args(args)
+
+
+def _ensure_nonce_args(args: Any):
+    """If nonce is not provided, ensure that recall_nonce is provided. If recall_nonce is provided, ensure that proxy is provided."""
+    if hasattr(args, "nonce") and args.nonce is None:
+        if not args.recall_nonce:
+            raise InvalidArgumentsError("Either --nonce or --recall-nonce must be provided")
+
+        if hasattr(args, "proxy") and not args.proxy:
+            raise InvalidArgumentsError("--proxy must be provided if --recall-nonce is used")
+
+
+def _ensure_receiver_args(args: Any):
+    """Ensure that receiver is provided."""
+    if hasattr(args, "receiver") and not args.receiver:
+        raise InvalidArgumentsError("--receiver must be provided")
+
+
+def _ensure_gas_limit_args(args: Any):
+    """Ensure that gas_limit is provided."""
+    if hasattr(args, "gas_limit") and not args.gas_limit:
+        if hasattr(args, "estimate_gas") and not args.estimate_gas:
+            raise InvalidArgumentsError("If --gas-limit not provided, --estimate-gas must be provided")
```

Comment:
But we will drop `--estimate-gas` in the near future, right?

Reply:
it's been already dropped in a future PR.